core/src/main/scala/org/http4s/QValue.scala

```diff
@@ -98,10 +101,29 @@ object QValue {
   def unsafeFromString(s: String): QValue =
     fromString(s).fold(throw _, identity)
 
+  private[http4s] val parser: Parser[QValue] = {
+    import cats.parse.Parser.{char => ch, _}
+    import cats.parse.Rfc5234._
+
+    val optWS = (crlf.rep.with1 ~ wsp.rep1).rep
+
+    val qValue = string1(ch('0') ~ (ch('.') ~ digit.rep1).rep)
+      .map(
```

Comment:
Can we use `.mapFilter` here and avoid throwing an exception?  It's new in cats-parse-0.2.

Reply:
Ok, I think  you mean something like that?
```
.mapFilter(
        QValue
          .fromString(_)
          .toOption
      )
```

Reply:
Yes, that looks right.  We want to avoid throwing exceptions, and `mapFilter` uses a selective functor, which is more efficient than flatMap in cats-parse.

Reply:
Done